modules/microshift-config-snippets.adoc

```diff
@@ -0,0 +1,166 @@
+At runtime, the YAML files inside `/etc/microshift/config.d` are merged into the existing {microshift-short} configuration, whether that configuration is a result of default values or a user-created `config.yaml`. You do not need to create a `config.yaml` to use a configuration snippet.
+
+Files in the snippet directory are sorted in lexicographical order and run sequentially. You can use numerical prefixes for snippets so that each is read in the order you want. The last-read file takes precedence when there is more than one YAML for the same parameter.
+
+[IMPORTANT]
+====
+Configuration snippets take precedence over both default values and a customized `config.yaml` configuration file.
+====
+
+[id="microshift-example-list-config-snippets_{context}"]
+== Example list configuration snippets
+
+Lists, or arrays, are not merged, they are overwritten. For example, you can replace a SAN or list of SANs by creating an additional snippet for the same field that is read after the first:
+
+.{microshift-short} configuration directory contents
+* `/etc/microshift/config.yaml.default` or `/etc/microshift/config.yaml`
+
+.Example {microshift-short} configuration snippet directory contents
+* `/etc/microshift/config.d/10-san.yaml`
+* `/etc/microshift/config.d/20-san.yaml`
++
+.Example `10-san.yaml` snippet
+[source,yaml]
+----
+apiServer:
+  subjectAltNames:
+    - host1
+    - host2
+----
++
+.Example `20-san.yaml` snippet
+[source,yaml]
+----
+apiServer:
+  subjectAltNames:
+    - hostZ
+----
++
+.Example configuration result
+[source,yaml]
+----
+apiServer:
+# ---
+  subjectAltNames:
+    - hostZ
+# ---
+----
+
+If you want to add a value to an existing list, you can add it to an existing snippet. For example, to add `hostZ` to an existing list of SANs, edit the snippet you have instead of creating a new one:
+
+.Example `10-san.yaml` snippet
+[source,yaml]
+----
+apiServer:
+  subjectAltNames:
+    - host1
+    - host2
+    - hostZ
+----
+
+.Example configuration result
+[source,yaml]
+----
+apiServer:
+# ---
+  subjectAltNames:
+    - host1
+    - host2
+    - hostZ
+# ---
+----
+
+[id="microshift-example-object-config-snippets_{context}"]
+== Example object configuration snippets
+
+Objects are merged together.
+
+.Example `10-advertiseAddress.yaml` snippet
+[source,yaml]
+----
+apiServer:
+  advertiseAddress: "microshift-example"
+----
+
+.Example `20-audit-log.yaml` snippet
+[source,yaml]
+----
+apiServer:
+  auditLog:
+    maxFileAge: 12
+----
+
+.Example configuration result
+[source,yaml]
+----
+apiServer:
+  advertiseAddress: "microshift-example"
+  auditLog:
+    maxFileAge: 12
+----
+
+[id="microshift-example-mixed-config-snippets_{context}"]
+== Example mixed configuration snippets
+
+In this example, the values of `advertiseAddress` and `auditLog.maxFileAge` are merged into the configuration, but only the `c.com` and `d.com` `subjectAltNames` values are retained because the numbering in the filename indicates that they are higher priority.
```

Comment:
```suggestion
In this example, the values of `advertiseAddress` and `auditLog.maxFileAge` fields are merged into the configuration, but only the `c.com` and `d.com` `subjectAltNames` values are retained because the numbering in the filename indicates that they are higher priority.
```